src/Nerdbank.Streams.Tests/SequenceTests.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Andrew Arnott. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Moq;
+using Nerdbank.Streams;
+using Xunit;
+using Xunit.Abstractions;
+
+public class SequenceTests : TestBase
+{
+    public SequenceTests(ITestOutputHelper logger)
+        : base(logger)
+    {
+    }
+
+    [Fact]
+    public void Empty()
+    {
+        var seq = new Sequence<byte>();
+        ReadOnlySequence<byte> ros = seq;
+        Assert.True(ros.IsEmpty);
+    }
+
+    [Fact]
+    public void Acquire_Sizes()
+    {
+        var seq = new Sequence<char>();
+
+        var mem1 = seq.GetMemory(16);
```

Comment:
What happens when you GetMemory multiple ties without advancing? You just get the last segment?

Reply:
If the last segment has enough slack space you just get that. Otherwise if the last segment was completely unused it gets replaced by a new segment. If the last segment had data then a new segment is appended and returned.